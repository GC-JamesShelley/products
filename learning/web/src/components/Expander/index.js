import React, { useState } from "react"
import styled from "styled-components"
import { rhythm } from "../../utils/typography"
import { mhraBlue, white } from "../../utils/colors"

const ExpanderStyled = styled.div`
  margin-bottom: ${rhythm(0.5)};
`

const Button = styled.button`
  color: ${white};
  font-weight: normal;
  font-size: 0.875rem;
  background-color: ${mhraBlue};
  padding: 0.5em 1em;
  border: 0;
  border-radius: 0.25rem;
  cursor: pointer;
`

const Body = styled.div`
  display: block;
  padding: ${rhythm(1)};
  &.hidden {
    display: none;
  }
`

const Expander = ({ title, children }) => {
  const [open, setOpen] = useState(false)

  const toggleOpen = e => {
    e.preventDefault()
    setOpen(!open)
  }
  return (
    <ExpanderStyled>
      <Button onClick={toggleOpen}>{title}</Button>
      <Body className={open ? undefined : "hidden"}>{children}</Body>
    </ExpanderStyled>
  )
}
export default Expander
