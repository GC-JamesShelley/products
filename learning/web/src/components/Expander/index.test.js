import React from "react"
import Expander from "./index"
import { shallow } from "enzyme"

describe(Expander, () => {
  it("should render", () => {
    const component = shallow(<Expander title={"Hola"} children={<></>} />)
    expect(component).toMatchSnapshot()
  })
})
