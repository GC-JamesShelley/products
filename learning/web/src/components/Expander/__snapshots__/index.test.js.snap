// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expander should render 🍪 1`] = `
<styled.div>
  <styled.button
    onClick={[Function]}
    title="expand"
  >
    Hola
  </styled.button>
  <styled.div
    className="hidden"
  >
    <styled.button
      onClick={[Function]}
      title="close"
    >
      <MdClose />
    </styled.button>
  </styled.div>
</styled.div>
`;
